torch/fx/graph.py

```diff
@@ -1854,10 +1854,14 @@ def forward(self, x):
         # DCE below will not behave as expected.
         self.lint()
 
+        impure_random = True
+        if torch._guards.TracingContext.try_get():
```

Comment:
Maybe we should store the compiler backend in the tracing context, and only do this for inductor.. thoughts @bdhirsh ?